\documentclass[a4paper]{article}

%\usepackage[british,UKenglish,USenglish,english,american]{babel}
\usepackage[none]{hyphenat}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{tabularx,ragged2e,booktabs,caption}
\usepackage{amsmath}
\usepackage{float}
\usepackage{url}
\usepackage{enumitem}


\usepackage[linktoc=all]{hyperref}
\usepackage{titlesec}
\hypersetup{
    colorlinks,
    citecolor=black,
    filecolor=black,
    linkcolor=black,
    urlcolor=black
}

\begin{document}
\SweaveOpts{concordance=TRUE}

%\VignetteIndexEntry{psygenet2r: Case study on GWAS on bipolar disorder}
\title{\texttt{psygenet2r}: Case study on GWAS on bipolar disorder}

\author{Alba Gutierrez-Sacristan \and Carles Hernandez-Ferrer \and Juan R. Gonzalez \and Laura I. Furlong}

\maketitle
\tableofcontents
\newpage

\section{Introduction}

\noindent Psychiatric disorders have a great impact on morbidity and mortality \cite{Murray, Whiteford}. According to the World Health Organization (WHO), one of every four people will suffer mental or neurological disorders \cite{WHO2001}. It has been suggested that most psychiatric disorders display a strong genetic component \cite{Issler, Levinson, Schizophrenia}. During the last years there has been a growing research in psychiatric disorders' genetics \cite{Sullivan}, and therefore the number of publications that focus on psychiatric disorders have increased steadily (Figure \ref{pubmed1}).\vspace{1cm}

\begin{figure}[H]
\begin{center}
<<pubmedArticles, echo=FALSE, fig=TRUE, height=8, width=8>>=
library(ggplot2)

data.file <- system.file(
  paste0("extdata", .Platform$file.sep, "psychiatricDisordersPubmed.csv"), 
  package="psygenet2r"
)

pmid <- read.delim(data.file, header=TRUE, sep=",")
pmid <- pmid[pmid$year<2015 & pmid$year>1950,]

pmid$year <- factor(pmid$year)
labels <- as.integer(seq(1950, 2015, by=5))

p <- ggplot(pmid, aes ( x = year, y = count ) ) +
    geom_bar ( stat = "identity", fill = "grey" ) +
    labs ( title = "Number of publications for psychiatric disorders in PubMed" , x = "year", y = "# of pmids") +
    theme_classic( ) + 
    scale_x_discrete(breaks=labels, labels=as.character(labels))+
    theme( plot.margin = grid::unit ( x = c ( 5, 15, 5, 15 ), units = "mm" ),
                                     axis.line = element_line ( size = 0.7, color = "black" ), 
           text = element_text ( size = 14 ) ,
           axis.text.x = element_text ( angle = 45, size = 11, hjust = 1 ) )

p
@
\caption{Number of publications for psychiatric disorders in PubMed. It has been obtained querying {\small \texttt{'psychiatric disorder[Title/Abstract]'}} from 1955 to 2014.}
\label{pubmed1}
\end{center}
\end{figure}\vspace{1cm}

\noindent However, there is still limited understanding on the cellular and molecular mechanisms leading to psychiatric diseases, which has limited the application of this wealth of data in the clinical practice. This situation also applies to psychiatric comorbidities. Some of the factors that explain the current situation is the heterogeneity of the information about psychiatric disorders and its fragmentation into knowledge silos, and the lack of resources that collect these wealth of data, integrate them, and supply the information in an intuitive, open access manner to the community. PsyGeNET has been developed to fill this gap. \texttt{psygenet2r} has been developed to facilitate statistical analysis of PsyGeNET data, allowing its integration with other packages available in R to develop data analysis workflows.\\

\noindent \texttt{psygenet2r} package allows to retrieve the genes associated to psychiatric diseases, or explore the association between a disease of interest and PsyGeNET diseases based on shared genes. In addition, \texttt{psygenet2r} allows the annotation of genes with psychiatric diseases based on expert-curated information. This functionality can be of interest to interpret the results of GWAS or Whole Exome Sequencing studies, in which a list of gene variants is obtained and there is a need to prioritize them based on their functional and clinical relevance. In this context, it would be of interest to know if there is information on their implication in psychiatric diseases. In this Case study we will describe how we can analyze the genes identified in a GWAS study in the context of psychiatric diseases using \texttt{psygenet2r}. For this purpose, we will use as an example the data obtained from a GWAS study on bipolar disorder published by \cite{McCarthy}. In this study, the authors analyzed the brain expression of 58 genes, previously identified in a GWAS of bipolar disorder, and correlated this information with structural MRI studies to identify brain regions that are abnormal in bipolar disorder. We will use this list of 58 genes from the bipolar disorder study to show the functionality of \texttt{psygenet2r} package.\\
% From these SNPs, genes in which they are located can be determined. \texttt{psygenet2r} allows the association analysis of these genes with others related to psychiatric disorders.\\ 

% \noindent According to the National Comorbidity Study, bipolar disorder (BD) presents a prevalence of nearly 4\% and is more common in women than men, being 25 years the median age of onset for BD \cite{Psychiatry}. Studies on BD suggest that the disease is consequence of multiple causes, including genetic variation \cite{McGuffin}. 

% \noindent One of the latest studies related to bipolar disorder is the one of McCarthy et al. \cite{McCarthy}. In their study, McCarthy et al., try to identify abnormal brain regions in patients with bipolar disorder. They mapped 58 gene expression signatures that could be related to BD according to the relationship of SNPs. They gathered information from various sources, including a meta-analysis of GWAS in BD \cite{ODZ4}. The top 100 SNPs associated with BD were selected and those SNPs that were associated with a single gene were identified resulting in a list of 58 genes. This list will be used in this case-study to show the functionality of \texttt{psygenet2r} package.\\ 


\subsection{Objective}

\noindent The goal of the study is to analyze a set of genes discovered by GWAS in the context of PsyGeNET. More specifically, we want to answer the following questions:\\
\begin{enumerate}
\item Are bipolar disorder genes associated to Substance Use Disorders according to PsyGeNET?
\item What is the level of evidence of these associations?
\item What is the function of the proteins encoded by these genes related to bipolar disorder?
\item Is bipolar disorder associated to major depression, cocaine use disorder or alcohol use disorder?
\end{enumerate}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Implementation}

\subsection{\texttt{psygenet2r} package}
\noindent PsyGeNET, a knowledge resource for the exploratory analysis of psychiatric diseases and their genes, focused on three psychiatric disorders: Major Depression (MD), Cocaine Use Disorder (CUD), Alcohol Use Disorder (AUD). PsyGeNET database is the result of data integration from DisGeNET and data extracted from the literature by text mining, followed by expert's curation \cite{psygenet}. The current version of PsyGeNET contains 2642 associations between 1271 genes and 37 psychiatric disease concepts. \texttt{psygenet2r} package contains functions to query and analyze PsyGeNET data, and to integrate with other information, as exemplified in this case study.\\

\subsection{Installation}
\noindent \texttt{psygenet2r} package is provided through Bioconductor \cite{bioc}. To install \texttt{psygenet2r} the user must type in the two following commands in R session:

<<bioC, eval=FALSE>>=
source( "http://bioconductor.org/biocLite.R" )
biocLite( "psyGeNET2R" )
@

<<load_library, eval = TRUE>>=
library( psygenet2r )
@

\section{Questions that can be answered using \texttt{psygenet2r}}

\noindent The first step that has to be done before doing any analysis is saving the genes in an R vector. For this case-study the 58 genes obtained from McCarthy et al. \cite{McCarthy} are saved into a vector called \texttt{genesOfInterest}.\\

\noindent Genes can be identified using the NCBI gene identifier or the Official Gene Symbol from HUGO.

<<genes>>=
genesOfInterest <- c("ADCY2", "AKAP13", "ANK3", "ANKS1A", 
"ATP6V1G3", "ATXN1", "C11orf80", "C15orf53", "CACNA1C", 
"CACNA1D", "CACNB3", "CROT", "DLG2", "DNAJB4", "DUSP22", 
"FAM155A", "FLJ16124", "FSTL5", "GATA5", "GNA14", "GPR81", 
"HHAT", "IFI44", "ITIH3", "KDM5B", "KIF1A", "LOC150197", 
"MAD1L1", "MAPK10", "MCM9", "MSI2", "NFIX", "NGF", "NPAS3", 
"ODZ4", "PAPOLG", "PAX1", "PBRM1", "PTPRE", "PTPRT", 
"RASIP1", "RIMBP2", "RXRG", "SGCG", "SH3PXD2A", "SIPA1L2",
"SNX8", "SPERT", "STK39", "SYNE1", "THSD7A", "TNR", 
"TRANK1", "TRIM9", "UBE2E3", "UBR1", "ZMIZ1", "ZNF274")
@

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection{How many of these genes are in PsyGeNET?}

\noindent In order to know how many of the genes of interest are present in PsyGeNET, \texttt{psygenetGeneList} function is used. This function requires as input the genes' vector and the selected database. For this analysis \texttt{"ALL"} databases are selected.\\ 

<<search_multiple>>=
m1 <- psygenetGene(
    gene     = genesOfInterest, 
    database = "ALL",
    verbose  = FALSE,
    warnings = FALSE
)
m1
@

\noindent The output is a \texttt{DataGeNET.Psy} object. It contains all the information about the different diseases associated with the genes of interest retrieved from PsyGeNET. By looking at the \texttt{DataGeNET.Psy} object, it can be observed that, according to PsyGeNET and by querying in \texttt{ALL} databases, 21 of the initial genes are found in PsyGeNET. These genes appear associated with 10 different disorders, involving a total of 37 gene-disease associations (GDAs).\\

% \noindent PsyGeNET is a comprehensive but not complete database that is filled with  data curated by experts. This is a challenging job to maintain up to date. The current release contains data from 1980 to 2013. For hence, some genes could not be found in PsyGeNET since their discovery is further 2013. In this example the genes from a reference article published in 2014 are used. From the total of 58 genes, 37 are not present in PsyGeNET.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection{Which diseases are associated to these genes according to PsyGeNET?}

\noindent In order to visualize the 38 GDAs between the 21 genes found in PsyGeNET and the 10 different disorders, \texttt{psygenet2r} provides several options. One of them is the GDA network, which can be obtained by applying the \texttt{plot} function to the \texttt{DataGeNET.Psy} object (\texttt{m1}), obtained from \texttt{psygenetGene} function (section 4.1). In the GDA network, green nodes represent diseases and orange nodes represent genes.\\

\begin{figure}[H]
\begin{center}
<<plot_disease_m, fig=TRUE, height=8, width=8>>=
plot( m1 )
@
\caption{Gene-Disease Association Network}
\label{gene-disease}
\end{center}
\end{figure}

\noindent As shown in Figure \ref{gene-disease}, most of the genes are associated to bipolar disorder, in agreement with McCarthy et al. \cite{McCarthy}, but some of these genes are also associated to substance related disorders and different depression sub-types.\\

\noindent We can also visualize the gene-disease associations at the level of psychiatric disorder class using heat-maps.\\

\noindent In the example shown in the next figure (Figure \ref{gene-psy}), the genes associated to the main disorder classes in PysGeNET with a PsyGeNET score larger than 0.1 are shown.  Eight from the 21 genes are associated with PsyGeNET diseases with a score greater than 0.1. ANK3 gene is the one with the highest mean score, approx. 0.6, and is associated to major depression class. Only one gene, NGF, passes the cut off in the cocaine use disorder group. On the other hand, none of the genes are associated to alcohol use disorders with a score higher than 0.1.\\


\noindent The heat-map can be obtained using the \texttt{plot} function and by setting the \texttt{type} argument to \texttt{"heatmap"} or \texttt{"heatmapScore"}. The cut-off argument can be added with the minimum score mean to be shown in the resulting graphic.\\



\begin{figure}[H]
\begin{center}
<<plot_psychiatric_m1Score, fig=TRUE, height=8, width=8>>=
plot( m1, type = "heatmapScore", cutOff = 0.1 )
@
\caption{Gene-Psychiatric disorder association heat-map according to the PsyGeNET mean score.}
\label{gene-psy}
\end{center}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection{What are the functions of the proteins encoded by these genes?}

\noindent \texttt{psygenet2r} package can be used to analyze gene attributes such as the panther class. The \textsl{PANTHER Protein Class Ontology} includes commonly used classes of protein functions. The panther class to which the genes belong according to their psychiatric disorder can be analyzed using the function \texttt{pantherGraphic}.\\ 

\noindent \texttt{pantherGraphic} function requires as input the list of genes (\texttt{genesOfInterest} vector) and the database (for instance \texttt{ALL}). The output of  \texttt{pantherGraphic} function is a bar-plot with the different panther classes in the Y-axis and the percentage of genes in the X-axis, grouped by PsyGeNET psychiatric disorders.\\

\begin{figure}[H]
\begin{center}
<<ggplot_panther, fig=TRUE, height=8, width=8>>=
pantherGraphic( genesOfInterest, "ALL" )
@
\caption{Panther graphic analysis of the genes of interest.}
\label{panther}
\end{center}
\end{figure}

\noindent The bar-plot shown in Figure \ref{panther} is obtained from the gene-list of interest. All the genes in the list that are associated with the psychiatric disease class of cocaine use disorders are signaling molecules. On the other hand, it can be observed that those genes associated with alcohol use disorder are 25\% receptors and 25\% phosphatases. Finally, those genes associated with major depression present the highest range of possible panther class.\\

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection{What is the level of evidence for each GDA?}

\noindent In PsyGeNET, each GDA is ranked with the PsyGeNET score, that reflects the level of evidence for each association. We can use PsyGeNET score to visualize the level of evidence in a heat-map.\\

\noindent In order to obtain the heat-map, the \texttt{plot} function can be applied to the \texttt{DataGeNET.Psy} object (\texttt{m1}). The argument type must to be set to \texttt{"heatmapGenes"} and the score cut-off can also be determined by the user. Notice that from the more than 100 associations, only the 100 with the highest scores will be shown.\\

\noindent To visualize those GDAs with a PsyGeNET score > 0.3, the \texttt{cutOff} argument is set to 0.3. As a result a heat-map is obtained with genes in the X axis and disorders in the Y axis. The green rank color is proportional to the PsyGeNET score, being the darkest one the association with the highest score.\\  

\begin{figure}[H]
\begin{center}
<<heatmap_disease_m, fig=TRUE, height=8, width=8>>=
plot( m1, type="heatmapGenes", cutOff=0.3 )
@
\caption{Gene-Disease Association Heat-map for GDAs with PsyGeNET score greater than 0.3.}
\label{gene-disease-2}
\end{center}
\end{figure}


\noindent Figure \ref{gene-disease-2} shows that those GDA with score greater than 0.3 are related to major depression disorders, specially with bipolar disorder. The genes with the greatest score (approximately 0.8) are ANK3 and CACNA1C genes, which appear strongly associated with bipolar disorder. These results are in agreement with the ones obtained in a meta-analysis of GWAS where these genes were identified to contain the higher risk alleles to bipolar disorder \cite{ODZ4}. \\

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection{For the disorder of interest, how many PubMed id support each gene-disease associations?}

\noindent In addition to the PsyGeNET score, we can also inspect the number of publications that support a GDA. \texttt{psygenet2r} allows the visualization of this information in a bar-plot, using the \texttt{plot} function with the \texttt{disorder} argument to indicate the disease of interest, and the \texttt{type} argument set to \texttt{"barplot"}. Figure \ref{pubmed-2} shows an example, with the genes in the X-axis and the number of PNIDs in the y-axis.\\

\begin{figure}[H]
\begin{center}
<<pmids_barplot, fig=TRUE, height=8, width=8>>=
plot( m1, name="Bipolar Disorder", type="barplot")
@
\caption{PubMed Ids that report each gene association with bipolar disorder.}
\label{pubmed-2}
\end{center}
\end{figure}

\noindent The results show that the CACNAC1 gene is associated with bipolar disorder in more than 40 publications, followed by ANK3 gene with more than 30 publications. The rest of genes have been associated with bipolar disorder in less than 10 publications.\\

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection{What are the sentences that report the association between genes and the disease of interest?}

\noindent \texttt{psygenet2r} package also allows the extraction of the sentences and the pmids for each one of the GDAs for a particular disease. Two functions are required, \texttt{psygenetGeneSentences} and \texttt{extractSentences}. So, first, \texttt{psygenetGeneSentences} function is applied to \texttt{genesOfInterest} with \texttt{ALL} databases. A \texttt{DataGeNET.Psy} object is obtained.\\

<<sentences1_query>>=
m2 <- psygenetGeneSentences(
    geneList = genesOfInterest, 
    database = "ALL"
)
m2
@

\noindent Then, the \texttt{extractSentences} function is applied to the previous \texttt{DataGeNET.Psy} object and \texttt{disorder} argument is set to the disorder of interest, in this case, \texttt{"Bipolar Disorder"}. Notice that if the disorder name is used, it must be written as it appears in PsyGeNET, otherwise results will not be found. The result is a data frame that contains the gene, disease, original db, the pmid and the sentence. As an example the first pmids are shown.\\

<<sentences2_extraction>>=
sentences <- extractSentences( m2, 
  disorder = "Bipolar Disorder" )
head(sentences$PUBMED_ID)
@

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection{Is bipolar disorder significantly associated with other diseases?}

\noindent An interesting calculation is to know which diseases are similar to a target disease based on shared genes. Since PsyGeNET database contains information on genes associated to psychiatric diseases we can use it to estimate disease similarity. The Jaccard Index is a statistic used for comparing the similarity of two sets. In our case, these sets are the genes associated to each one of the target diseases. In the \texttt{psygenet2r} package, the Jaccard Index is calculated by using the function \texttt{jacacardEstimation}.\\

\noindent The function \texttt{jacacardEstimation} allows us to calculate the Jaccard Index using both PsyGeNET's data (like disease names or CUIs) and external information as vectors of genes. Moreover the Jaccard Index, this function calculates an associated p-value to the index.\\

\noindent The strategy to calculate the Jaccard Index and its p-value corresponds to:
\begin{enumerate}[label=(\Alph*)]
  \item Calculate the Jaccard Index between the pair of diseases. Let's call it \textit{rJI}.
  \item Randomly select a set of genes from DisGeNET for each one of the input diseases (or set of genes).
  \item See how many of these genes are in common and divide them for the total length to calculate their Jaccard Index. Let's call it \textit{iJI}.
  \item Calculate the p-value by dividing the count of the \textit{iJI} higher than the real \textit{rJI} by the number of attempts we performed the steps B and C plus one (\texttt{nboot + 1}).
\end{enumerate}

\noindent Let's calculate the Jaccard Index for the genes of interest and bipolar disorder:

<<jaccard_1>>=
xx <- jaccardEstimation( genesOfInterest, "Bipolar Disorder", 
  database = "CURATED", nboot = 500 )
xx
@

\noindent The result shows the number of interactions used to calculate the p-value and the type of input, in this case a list of genes and a disease. The Jaccard Index and the p-value can be extracted using the function \texttt{extract}:

<<jaccard_2>>=
extract( xx )
@

\noindent Now we have seen the Jaccard Index,and its p-value, between our genes of interest and bipolar disorder (\textit{JI}: 0.02; \textit{pval}: 0.002). The function \texttt{jaccardEstimation} also allows to calculate the Jaccard Index of our input, the list of genes, and all the diseases in PsyGeNET.

<<jaccard_3, cache=TRUE>>=
xx <- jaccardEstimation( genesOfInterest, 
  database = "CURATED", nboot = 500 )
@

\noindent The result from the Jaccard Index estimation can be plot using the function \texttt{plot}. The result is a bar-plot where the p-value of each comparison between our genes and PsyGeNET's disease is shown. A \texttt{cutOff} argument can be added in order to visualize only those diseases with an statistically significant p-value, in this case it is not necessary:

\begin{figure}[H]
\begin{center}
<<jaccard_4, fig=TRUE, height=5>>=
plot( xx )
@
\end{center}
\caption{Bar-plot where the Jaccard Index of each comparison between the list of genes of interest and PsyGeNET's diseases is shown.}
\label{jacc}
\end{figure}

\noindent The similarity between diseases according to shared genes can be visualized using Venn diagrams. \texttt{psygenet2r} package allows to plot two types of Venn diagrams: the conventional one and a variation of it. In the variation of the Venn diagram, a triangle is drawn, representing each vertex the three psychiatric disorders from PsyGeNET (MD, AUS and CUD). In the middle of the triangle, those genes that are in common for the three psychiatric disorders will appear, while in the vertex will be represent those that are exclusively to each one of them.\\

\noindent In order see the similarity between our genes of interest and PsyGeNET's psychiatric disorder, the \texttt{type} argument of \texttt{plot} function is set to \texttt{VennA}. This will allow to see how many of the 21 genes, that appear in PsyGeNET, are associated with each psychiatric disorder:\\

\begin{figure}[H]
\begin{center}
<<plot_venn_1_2, fig=TRUE, height=8, width=8>>=
plot( m1, type = "vennA" )
@
\caption{Venn Diagram: Genes shared between the three main psychiatric disorders}
\label{vennA}
\end{center}
\end{figure}

\noindent Figure \ref{vennA} clearly shows that none of our genes of interest are associated to the three disorders. There are 14 exclusively associated to major depression and 4 to alcohol use disorder. The gene that appears associated to cocaine use disorder is also associated to major depression. Alcohol use disorder and major depression share two genes.\\

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\newpage
%\section {Bibliography}

\begin{thebibliography}{9}
\bibitem{Murray}  % [6]
  Murray Christopher J.L., Lopez Alan D.
  \textbf{Measuring the global burden of disease}.
  N.Engl. J. Med.
  2013.
  doi:10.1056/NEJMra1201534
\bibitem{Whiteford} % [11]
  Whiteford Harvey A., Degenhardt Louisa, Rehm Jürgen, Baxter Amanda J., Ferrari Alize J., Erskine Holly E., et al.
  \textbf{Global burden of disease attributable to mental and substance use disorders: findings from the Global Burden of Disease Study 2010}.
  Lancet
  2013.
\bibitem{WHO2001}
  World Health Organization.
  \textbf{The world health report 2001 - Mental Health: New Understanding, New Hope}.
  ISBN 92-4-156201-3
\bibitem{Issler}  % [2]
  Issler Orna, and Chen Alon.
  \textbf{Determining the role of microRNAs in psychiatric disorders}. 
  Nature Reviews Neuroscience
  2015.
  doi:10.1038/nrn3879
\bibitem{Levinson}  %[3]
  Levinson Douglas F., Mostafavi Sara, Milaneschi Yuri, Rivera Margarita, Ripke Stephan, Wray Naomi R., Sullivan Patrick F.
  \textbf{Genetic studies of major depressive disorder: why are there no genome-wide association study findings and what can we do about it?}
  Biol. Psychiatry
  2014.
  doi:http://dx.doi.org/10.1016/j.biopsych.2014.07.029
\bibitem{Schizophrenia} %[9]
  Schizophrenia Working Group of the Psychiatric Genomics Consortium. 
  \textbf{Biological insights from 108 schizophrenia-associated genetic loci}.
  Nature
  2014.
  doi:10.1038/nature13595
\bibitem{Sullivan}  % [10]
  Sullivan Patrick F., Daly Mark J., O'Donovan Michael.
  \textbf{Genetic architectures of psychiatric disordlers: the emerging picture and its implications}. 
  Nat. Rev. Genet.
  2012
  doi:10.1038/nrg3240.
\bibitem{psygenet}  % [1]
  Alba Gutiérrez-Sacristán, Solène Grosdidier, Olga Valverde, Marta Torrens, Àlex Bravo, Janet Piñero, Ferran Sanz, Laura I. Furlong.
  \textbf{PsyGeNET: a knowledge platform on psychiatric disorders and their genes}.
  Bioinformatics
  2015.
  doi:10.1093/bioinformatics/btv301
\bibitem{Psychiatry}  % [12]
  Black Donald W., Andreasen Nancy C.
  \textbf{Introductory Textbook of Psychiatry (5th ed.)}.
  American Psychiatric Publishing, Inc., pg. 158
  ISBN-13: 978-1585624003
\bibitem{McGuffin}  % [5]
  McGuffin Peter, Rijsdijk Fruhling, Andrew Martin, Sham Pak, Katz Randy, Cardno Alastair. 
  \textbf{The heritability of bipolar affective disorder and the genetic relationship to unipolar depression}.
  JAMA Psychiatry
  2003.
  doi:10.1001/archpsyc.60.5.497
\bibitem{ODZ4}      %[8]
  PGCBD Consortium.
  \textbf{Large-scale genome-wide association analysis of bipolar disorder identifies a new susceptibility locus near ODZ4}. 
  Nat Genet
  2011.
  doi:10.1038/ng.943.
\bibitem{bioc}
  Bioconductor
  \url{http://www.bioconductor.org}
\bibitem{McCarthy}  % [4]
  McCarthy Michael J., Liang Sherri, Spadoni Andrea D., Kelsoe John R., Simmons Alan N.
  \textbf{Whole brain expression of bipolar disorder associated genes: structural and genetic analyses}.
  PLoS One
  2014.
  doi:10.1371/journal.pone.0100204
%\bibitem{Pettinati} %[7]
%  Pettinati Helen M., O’Brien Charles P.,  Dundon William D.
%  \textbf{Current status of co-occurring mood and substance use disorders: a new therapeutic target}. 
%  Am. J. Psychiatry
%  2013.
%  doi:http://dx.doi.org/10.1176/appi.ajp.2012.12010112
\end{thebibliography}

\end{document}
