\documentclass[a4paper]{article}

%\usepackage[british,UKenglish,USenglish,english,american]{babel}
\usepackage[none]{hyphenat}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{tabularx,ragged2e,booktabs,caption}
\usepackage{amsmath}
\usepackage[left=3cm,right=3cm,top=3cm,bottom=3cm]{geometry}
\usepackage{float}
\usepackage[linktoc=all]{hyperref}
\usepackage{titlesec}
\hypersetup{
    colorlinks,
    citecolor=blue,
    filecolor=blue,
    linkcolor=blue,
    urlcolor=blue
}

\begin{document}

\newcommand{\subsubsubsection}[1]{\vspace{0.15cm}\noindent\textbf{#1}\vspace{0.15cm}}

%\VignetteIndexEntry{psygenet2r: An R package for querying PsyGeNET and to perform comorbidity studies in psychiatric disorders}
\title{\texttt{psygenet2r}: An R package for querying PsyGeNET and to perform comorbidity studies in psychiatric disorders}

\author{Alba Gutierrez-Sacristan \and Carles Hernandez-Ferrer \and Juan R. Gonzalez \and Laura I. Furlong}

\maketitle
\tableofcontents
\newpage

\section{Introduction}

The \texttt{psygenet2r} package contains functions to query PsyGeNET \cite{REF1}, a resource on psychiatric diseases and their  genes. The \texttt{psygenet2r} package includes analysis functions to study psychiatric diseases, their genes and disease comorbidities. A special focus is made on visualization of the results, providing a variety of representation formats such as networks, heatmaps and barplots.

\subsection{Background}

\noindent During the last years there has been a growing interest in the genetics of psychiatric disorders, leading to a concomitant increase in the number of publications that report these studies \cite{REF2}. However, there is still limited understanding on the celular and molecular mechanisms leading to psychiatric diseases, which has limited the application of this wealth of data in the clinical practice. This situation also applies to psychiatric comorbidities. Some of the factors that explain the current situation is the heterogeneity of the information about psychiatric disorders and its fragmentation into knowledge silos, and the lack of resources that collect these wealth of data, integrate them, and supply the information in an intuitive, open access manner to the community. PsyGeNET has been developed to fill this gap. \texttt{psygenet2r} has been developed to facilitate statistical analysis of PsyGeNET data, allowing its integration with other packages available in R to develop data analysis workflows.\\

\noindent PsyGeNET is a resource for the exploratory analysis of psychiatric diseases and their associated genes. It is focused on eight psychiatric disorders: alcohol use disorders, bipolar disorders and related disorders, depressive disorders, schizophrenia spectrum and other psychotic disorders, cocaine use disorders, substance/drug induced depressive disorder, cannabis use disorders and drug-induced psychosis. PsyGeNET allows the exploration of the molecular basis of psychiatric disorders by providing a comprehensive set of genes associated to each disease. Moreover, it allows the analysis of the molecular mechanisms underlying psychiatric disease comorbidities. 
 PsyGeNET database is the result of the data extracted from the literature by text mining using BeFree \cite{REF4}, followed by manual curation by domain experts. A team of 22 experts of different fields in psychiatry and neurosciences, defined each psychiatric disorder in terms of Concept Unique Identifiers (CUI) from the Unified Medical Language System (UMLS) and reviewed the association.The current version of PsyGeNET (v02) contains 3,771 associations, between 1,549 genes and 117 psychiatric disease concepts.\\

\noindent With \texttt{psygenet2r} package the user will be able to submit queries to PsyGeNET from R, perform a variety of analysis on the data, and visualize the results through different types of graphical representations.\\

\noindent The tasks that can be performed with \texttt{psygenet2r} package are the following:
\begin{enumerate}
\item Retrieve Gene-Disease Associations (GDAs) from PsyGeNET using as query a gene or a disease (single or a set of genes/diseases) of interest
\item Visualize the results according to the GDAs' attributes: PsyGeNET score, number of publications, sentences that report the GDA, source dadatabase
\item Visualize the results accoring to the disease (disease class) or gene (Panther class) attributes
\item Analyze the association between two diseases from the molecular perspective (using the Jaccard index). 
\end{enumerate}

\noindent In the following sections the specific functions that can be used to address each of these tasks are presented.\\

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection{Installation}

\noindent The package \texttt{psygenet2r} is provided through Bioconductor. To install \texttt{psygenet2r} the user must type the two following commands in an R session:

<<bioC, eval=FALSE>>=
source( "http://bioconductor.org/biocLite.R" )
biocLite( "psygenet2r" )
@

<<load_library, eval=FALSE>>=
library( psygenet2r )
@

<<load_library_2, echo=FALSE>>=
suppressMessages(suppressWarnings(library( psygenet2r )))
@


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection{\texttt{DataGeNET.Psy object}}

\noindent \texttt{DataGeNET.Psy} object is obtained when \texttt{psygenetGene} and \texttt{psygenetDisease} functions are applied. This object is used as input for the rest of \texttt{psyGeNET2r} functions, like the \texttt{plot} function.\\

\noindent \texttt{DataGeNET.Psy} object contains all the information about the different diseases/genes associated with the gene/disease of interested retrieved from PsyGeNET. It object contains a summary of the search, such as the search input (gene or disease), the selected database, the gene or disease identifier, the number of associations found (N. Results) and the number of unique results obtained (U. Results).\\

<<dataGenet1, echo=FALSE>>=
t1 <- psygenetGene( gene = 4852, 
                    database = "ALL", 
                    check = FALSE)
@

<<dataGenet2>>=
t1
class(t1)
@

\noindent This object comes with a series of function to allows users to interact with the information retireved from PsyGeNET. These functions are \texttt{ngene}, \texttt{ndisease}, \texttt{extract} and \texttt{plot}. The first function \texttt{ngene} return the number of retrieved genes for a given query. \texttt{ndisease} is the homologous function but for the diseases. The function \texttt{extract} returns a formatted \texttt{data.frame} with the complete set of information downloaded from PsyGeNET. Finally, the \texttt{plot} function lets visualize the results in a variety of ways such as gene-disease association networks or heatmaps.\\

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\section{PsyGeNET and \texttt{psygenet2r}}

\noindent The PsyGeNET web interface can be explored by searching a specific gene or a specific disease, and \texttt{psygenet2r} package has the same options. Therefore, the starting point for \texttt{psygenet2r} are \texttt{psygenetGene} and \texttt{psygenetDisease} functions.\\

\noindent PsyGeNET data is classified according to the database used as a source of information ("source database"). Therefore, any query run on PsyGeNET requires to specify the source database using the argument called \texttt{database}. Table \ref{tab:psygenet_databases} shows the source databases in PsyGeNET and their description. By default, the database \texttt{ALL} is used in \texttt{psygenet2r}. For illustrating purposes along the vignette, database \texttt{ALL} will be used in most of code snippets.\\
\noindent \begin{minipage}{\linewidth}
\small{
  \centering
  \captionof{table}{Source databases included in PsyGeNET} \label{tab:psygenet_databases} 
  \begin{tabular}{|l|l|}
    \hline
Name             & Description                                                \\
\hline
\texttt{psycur15}  & Psychiatric disorders Gene association manually curated first release (2 experts)
\\
\texttt{psycur16}     & Psychiatric disorders Gene association manually curated second release (22 experts)
\\
\texttt{ALL}     & All previous Databases                                     \\
    \hline
  \end{tabular} \par
%\bigskip
}
\end{minipage}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Retrieve gene-disease associations (GDAs) from psygenet2r}

\subsection{Using genes as a query}
\texttt{psygenet2r} package allows exploring PsyGeNET information using a specifc gene or a list of genes. It retrieves the information that is available in PsyGeNET (associated diseases, source database, PsyGeNET score, number of publications, attributes of genes, etc) and allows to visualize the results in different  ways.


\subsubsection{Using as a query a single gene}

\noindent In order to look for a single gene into PsyGeNET, we can use the \texttt{psygenetGene} function. This function retrieves PsyGeNET's information using both, the NCBI gene identifier and the official Gene Symbol from HUGO. It contains also other arguments like the database to query, the PsyGeNET score and the check argument, that allows the user to decide if he wants to validate or not the gene before interrogating the database.\\
  
\noindent As an example, the gene \textit{NPY}, whose entrez id is \textsl{4852} is queried using \texttt{psygenetGene} function, and using alternatively the official HUGO Gene Symbol. In this example database \texttt{"ALL"}.\\

<<search_gene_1, cache=TRUE>>=
t1 <- psygenetGene( gene = 4852, 
                    database = "ALL", 
                    check = FALSE)
t1
@


<<search_gene_2, cache=TRUE>>=
t2 <- psygenetGene( gene = "NPY", database = "ALL", check = FALSE )
t2
@

\noindent Both cases result in an \texttt{DataGeNET.Psy} object:

<<searcg_gene_class>>=
class(t1)
class(t2)
@

\noindent In the particular example used, by inspecting the \texttt{DataGeNET.Psy} object, we can see that the gene \textit{NPY} is associated to 13 different diseases in PsyGeNET (with no restriction on the PsyGeNET score). 

\subsubsection{Ploting the results of a Single Gene Query}

\noindent \texttt{psygenet2r} offers several options to visualize the results from PysGeNET: a  network showing the diseases related to the gene of interest, or a network showing the strength of the association between the three main psychiatric disorders in PsyGeNET and the gene of interest. Each one of these graphics can be obtained changing the type argument.\\ 

\noindent By default, \texttt{psygenet2r} shows this type of network when ploting a \texttt{DataGeNET.Psy} object obtained by a gene-query. The result is a network where green nodes are diseases and the orange node is the gene of interest.\\

\begin{figure}
\begin{center}
<<plot_disease>>=
plot( t1, type = "individual disease" )
@
\end{center}
\end{figure}

\noindent On the other hand, results can be visualized according to the 3 psychiatric disorders classes available in PsyGeNET (major depression, alcohol use disorder and cocaine use disorder) setting the \texttt{type} argument to \texttt{"disease class"}. As a result, a network with 4 nodes is obtained. The node's size of each psychiatric disorder is proportional to the number of disease concepts that belongs to each disease class, from the total number of diseases associated to the gene.\\

\begin{figure}[H]
\begin{center}
<<plot_psychiatric>>=
plot( t1, type = "disease class" )
@
\end{center}
\end{figure}

\noindent In our example, NPY is associated to the three psychiatric disorders, with an important contribution of major depression disorders.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsubsection{Using as a query a list of genes}

\noindent In the same way, \texttt{psygenet2r} allows to query PsyGeNET given a list of genes of interest. The same function, \texttt{psygenetGene}, accepts a vector of NCBI gene identifiers or HUGO official gene symbols.\\

\noindent To illustrate this functionality, a list of 20 genes was extracted from the article entitled \textsl{"The Genetics of Major Depression"} \cite{REF5}, where these genes are associated to depression. The vector of genes can be defined as follows:

<<search_multiple_genes>>=
genesOfInterest <- c( "COMT", "CLOCK", "DRD3", "GNB3", "HTR1A",
                      "MAOA", "HTR2A","HTR2C", "HTR6", "SLC6A4", 
                      "ACE",  "BDNF", "DRD4", "HTR1B", "HTR2B",
                      "HTR2C", "MTHFR", "SLC6A3", "TPH1", "SLC6A2", 
                      "GABRA3"
)
@

\noindent Then, the function \texttt{psygenetGene} is applied. In this case an extra argument called \texttt{verbose} was set to \texttt{TRUE}. This shows some information during the query-process, for example the message informing that there are repeated genes in the list (the gene \textsl{HTR2C} was placed twice in the list to raise this message).\\

<<search_multiple_search>>=
m1 <- psygenetGene(
    gene     = genesOfInterest, 
    database = "ALL",
    verbose  = TRUE,
    check = FALSE
)
@

\noindent A \texttt{DataGeNET.Psy} object is obtained. In this particular example, 20 genes are present in PsyGeNET and are associated to 34 diseases, involving 238 GDAs.\\

<<show_multiple>>=
m1
@

\subsubsection{Ploting the results of the query using a list of genes}

\noindent \texttt{psygenet2r} provides several options to visualize the results of these queries, such as networks, heatmaps and venn diagrams.\\ 

\noindent As for single gene, the default option in \texttt{psygenet2r} results in a network chart, where the green nodes belong to diseases and the oranges nodes belong to genes. It is also possible to visualize the results by grouping the diseases according to the psychiatric disorders present in PsyGeNET specifying the argument type.\\

%' <<plot_psychiatric1, fig=TRUE, height=8, width=8>>=
%' plot( m1 )
%' @
%' 
%' <<plot_psychiatric2, fig=TRUE, height=8, width=8>>=
%' plot( m1, type = "disease class" )
%' @

\begin{figure}[H]
\begin{center}
<<plot_psychiatric1, fig.widtgh=10>>=
layout( matrix( c( 1, 2 ), nrow = 1 ) )
plot( m1 )
plot( m1, type = "disease class" )
@
\end{center}
\end{figure}

\noindent \texttt{psygenet2r} package allows to visualize the GDAs attributes in a heatmap. The argument \texttt{type} must be \texttt{"heatmapGenes"} and the PsyGeNET score can also be determined by the user setting the \texttt{cut-off} argument to the score of interest. In this example, the cut-off is set to 0.5, it means that only those associations with an score equal or higher to 0.5 will be shown. \textbf{Warning}: if there are more than 100 associations, only the first 100 with highest score are shown in the heatmap. In this kind of representation we can identify genes that are associated to several diseases (e.g. SLC6A4) and also genes that are specific for a particular diseases (e.g. GABRA3).

<<heatmap_disease_m, fig.width=8, fig.height=8 >>=
plot( m1, type="heatmapGenes" )
@


\noindent \texttt{psygenet2r} package also allows to analyze a gene list according to the function of the proteins encoded by these genes. The PANTHER Protein Class Ontology classifies proteins according to their function.\\

 
\noindent The \texttt{pantherGraphic} function shows the Panther class to which the proteins belong according to their associated psychiatric disorder. It provides a graphic with the results of these analysis, being the input a list of genes and the database (\texttt{ALL}, \texttt{CURATED}, \texttt{PsyCUR}, etc). The input genes can be from a vector that contains the genes of interest, or from the genes obtained in the \texttt{DataGeNET.psy} object in a disease or disease-list query. An score argument can be added to filter results. It can also be done given a DataGeNET.psy object obtained by querying with a single gene.


<<panther_gene>>=
genesOfInterest <- unique( genesOfInterest )
pantherGraphic( genesOfInterest, "ALL", check = FALSE )
@

\subsubsection{Ploting results of the Multiple Gene Query}

\noindent In addition, the results of a query using a list of genes can be visualized using barplots and pie charts.\\

\noindent Here, in this first barplot we show how many of the genes that are associated with a disease category are exclusively associated to it and how many of them are also associated with any other category.\\

<<plot_diseaseBarplot, fig.height=8, fig.width=8>>=
<<<<<<< HEAD
=======
plot( m1, type = "diseaseBarplot" )
>>>>>>> 67b8c39f12f2be9787a889bf2ecd1783ecac5aae
geneAttrPlot( m1, type = "category" )
@

\noindent If we are only interested in how many of our input genes are associated to each disease category, a pie chart can be obtained applying the \texttt{plot} function and setting up the \texttt{type} argument to \texttt{pieChart}.\\

<<plot_pieChart, fig.height=8, fig.width=8>>=
geneAttrPlot( m1, type = "pie" )
@

\noindent On the other hand, this second barplot show us the number of cuis and disease categories associated to each one of our genes of interest.\\

<<plot_geneBarplot, fig.height=8, fig.width=8>>=
geneAttrPlot( m1, type = "gene" )
@

\noindent An alternative graphical representation is a barplot showing gene-disease association according to each disease category taking into account the type of evidence of each assocation (association, no association, both).\\

<<barplotIP, fig.height=8, fig.width=8>>=
geneAttrPlot( m1, type = "index" )
@

\noindent The other one allows to analyze if the genes that are being studied present a specific association with a subtype of disorder or if they are associated with several of them in the same psychiatric disorder. The percentage of diseases to which a gene appear associated with each psychiatric disorder is estimated. This percentage is relative to the total possible subtypes of disorders present in PsyGeNET (3 for cocaine use disorder, 7 for alcoholism use disorder and 27 for major depression). The resultant values are represented in a heatmap according to a blue color scale.\\

\noindent In order to obtain this graphic, \texttt{type} must be set to \texttt{"heatmap"}. The resulting heatmap shows which are the genes that are higher or lower associated to each one of the psychiatric disorders. 

<<plot_psychiatric_heamap>>=
plot( m1, type = "heatmap" )
@

\noindent As it could be expected for the genes that are used as genes of interest, all of them are associated to major depression, to a greater or lesser extent. Some of them are also associated to the other two psychiatric disorders.  



\subsubsection{Enrichment analysis for a list of genes}

\noindent The R package \texttt{psygenet2r} allows to perform an enrichment analysis on a list of genes with PsyGeNET diseases. It is done by using the function \texttt{enrichedPD}. In order to illustrate this function, a set of 8 genes extracted from a GWAS study \cite{Treutlein} was selected.\\


<<enrichment>>=
genesOfInterest <- c("PECR", "ADH1C", "CAST", "ERAP1", "PPP2R2B", 
                     "ESR1", "GATA4", "CDH13")
tbl <- enrichedPD( genesOfInterest, database = "ALL")
tbl
@

<<enrichment2>>=
genesOfInterest <- c(55825, 126, 831, 51752, 5521)
tb2 <- enrichedPD( genesOfInterest, database = "ALL")
tb2
@

\noindent \textbf{The result is a table with a p-value of the enrichment of the given list of genes for each psychiatric disorder in PsyGeNET. As we can see these genes are enriched in the alcohol use disorder genes of PsyGeNET (p-val = 0.0009). This result is in agreement to what Treutlein et al. find in their analysis, in which they determined that these genes where associated with alcohol dependence.}\\ 

\subsubsection{Enrichment analysis based on anatomical terms (TopAnat) for a list of genes}

<<topAnat, message=FALSE, results='hide'>>=
genesOfInterest <- c( "COMT", "CLOCK", "DRD3", "GNB3", "HTR1A",
                      "MAOA", "HTR2A","HTR2C", "HTR6", "SLC6A4", 
                      "ACE",  "BDNF", "DRD4", "HTR1B", "HTR2B",
                      "HTR2C", "MTHFR", "SLC6A3", "TPH1", "SLC6A2", 
                      "GABRA3"
)
tpAnat <- topAnatEnrichment( genesOfInterest, cutOff = 1 )
@

<<show_topAnat>>=
head( tpAnat )
@

\noindent \textbf{The result is a table with a p-value of the enrichment of the given list of genes for each psychiatric disorder in PsyGeNET. As we can see these genes are enriched in the alcohol use disorder genes of PsyGeNET (p-val = 0.0009). This result is in agreement to what Treutlein et al. find in their analysis, in which they determined that these genes where associated with alcohol dependence.}\\ 

\subsubsection{Sentences that report a GDA}

\noindent \texttt{psygenet2r} package also allows extract the pmids sentences that report a gene-disease asssociation. It is done by using two different functions, \texttt{psygenetGeneSentences} and  \texttt{extractSentences}. \texttt{psygenetGeneSentences} needs as input a gene list and a database to query in. The output of this function is a \texttt{DataGeNET.Psy} object. This object is passed to the \texttt{extractSentences} function, that also needs the disorder of interest.\\

<<gda_sentenceGene>>=
genesOfInterest
sss <- psygenetGeneSentences( geneList = genesOfInterest,
                             database = "ALL")
sss

geneSentences <- extractSentences( object = sss,
                                   disorder = "alcohol dependence")
dim(geneSentences)
@

\noindent The result is a data frame that contains the gene, disease, original db, the pmid and the sentence.\\ 

\subsection{Using diseases as a query}

\texttt{psygenet2r} package allows to explore PsyGeNET information searching a specifc disease or a list of diseases. As in the case of genes, it retrieves the information that is available in PsyGeNET and allows to visualize the results in several ways.

\subsubsection{Using as a query a single disease}

\noindent In order to look for a single disease into PsyGeNET, \texttt{psygenet2r} has the \texttt{psygenetDisease} function. This function allows you to obtain PsyGeNET's information using both disease id or disease name, and the database as input (by default is \texttt{CURATED}). \textbf{Warning}: if user uses the disease name, it must be written in the same way that appears in PsyGeNET, otherwise the disorder will not be found (Alba: i think this warning can be removed, since you provide a function to search for the standard name and chi of a disease).\\  

\noindent If the user do not know the disease identifier, it can use the \texttt{getUMLS} function to obtain disease names and UMLS CUIs from a string query. Providing as input the term and source of interest, \texttt{getUMLs} function retrieves all the PsyGeNET concepts that contain it. As an example it is shown the query results for \texttt{depressive} term in \texttt{ALL} databases.\\ 

<<getUMLS>>=
getUMLs("depressive", database="ALL")
@

\noindent As an example, the disease \textit{Single Major Depressive Episode}, whose disease id is \textsl{umls:C0024517} is queried using \texttt{psygenetDisease} function, and using both, disease name and disease id. For this example database \texttt{"ALL"} is selected:\\

<<search_diseaseId_1>>=
d1 <- psygenetDisease( disease  = "umls:C0024517", 
                       database = "ALL", 
                       score    = c('>', 0.5 ) )
d1
@

<<search_diseaseName_1>>=
d2 <- psygenetDisease( disease = "major depressive affective disorder, single episode, unspecified", 
                       database = "ALL",
                       score    = c('>', 0 ) )
d2
@

\noindent Both cases result in an \texttt{DataGeNET.Psy} object, that contains the same information as in the gene query search:

<<search_gene_class>>=
class(d1)
class(d2)
@


\noindent The argument \texttt{score} is filled with a vector which first position can be \texttt{'<'} or \texttt{'>'} to indicate if the threshold is read as lower or upper. The second argument is the threshold in itself which will always be included. This argument is also present in \texttt{psygenetGene}.

\subsubsection{Plotting results of a Single Disease Query}

\noindent \texttt{psygenet2r} package offers several options to visualize the results from PysGeNET given a disease: a network showing the genes related to the disease of interest and a barplot showing how many publications report each one of the gene-disease associations. \\

\noindent By default, \texttt{psygenet2r} shows the GDAs network when ploting a \texttt{DataGeNET.Psy} object with a disease-query. The result is a network where, orange nodes are genes and the central and green node is the disease of interest.\\  

<<plot_visualizing_single_disease_search>>=
plot ( d1 )
@

\subsubsection{Using a list of diseases as a query}

\noindent In the same way, \texttt{psygenet2r} allows to query PsyGeNET given a set of diseases of interest. The same function, \texttt{psygenetDisease}, accepts a vector of disease-names or disease-ids (umls code).\\

\noindent To illustrate this functionality, two disorders has been selected: bipolar disorder and major depressive disorder. The vector of diseases can be defined for example, as follows:\\

<<diseaseList>>=
diseasesOfInterest <- c( "manic depressive illness","major depression" )
@

<<search_diseases_1>>=
tt <- psygenetDisease( disease  = diseasesOfInterest,
                       database = "ALL" )
tt
@

<<search_diseases_2>>=
dm <- psygenetDisease( disease  = c( "umls:C0005586", "umls:C1269683" ), 
                       database = "ALL" )
dm
@

<<search_diseases_3>>=
tm <- psygenetDisease( disease  = c( "manic depressive illness","umls:C1269683" ), 
                       database = "ALL" )
tm
@

\noindent Three cases result in an \texttt{DataGeNET.Psy} object:

<<search_gene_class_2>>=
class(tt)
class(dm)
class(tm)
@

\noindent This type of object contains all the information about the different genes associated with the diseases of interest retrieved from PsyGeNET. By inspecting the \texttt{DataGeNET.Psy} object we can see that, according to PsyGeNET and querying in ALL databases, the 2 disorders of interest are associated to 494 different genes in 700 different associations. 

\subsubsection{Ploting results: Multiple Diseases}

\noindent \texttt{psygenet2r} provides a network graphic and a heatmap to visualize the results of search with multiple input items. \\

\noindent As for single disease, GDAs network is the default option in \texttt{psygenet2r}. In the resulting network chart, the green nodes represent diseases and the oranges nodes represent genes.\\ 

<<plot_disease_tm>>=
plot( tm )
@


\noindent Another possible option is visualize it in a heatmap. The argument \texttt{type} can be set to \texttt{"heatmap"} and the score argument must be determined by the user. For example, in the previous search, 494 different genes were found associated to the disorders of interest. The PsyGeNET score ranges from 0 to 1; greater score means that there are more evidences supporting the association, so we can restrict our heatmap to those association with an score greater or equal to 0.7. The darkest red belongs to the greatest score. 

<<heatmap_disease_tm>>=
plot( tm, type = "heatmap" )
@

\noindent The result is a heatmap where the genes are located at X axis, and disordes appear at Y axis. The red rank color is related to the PsyGeNET score, being the darkest one the association with the highest score. 

\subsubsection{Barplot according to number of publications that support the GDA}

\noindent \texttt{psygenet2r} package allows to see how many pubmed ids support each gene-disease association. This can be visualize in a barplot by determining the gene or disease id in the \texttt{name} argument and setting \texttt{type} argument to \texttt{"barplot"}.\\ 

<<barplot_visualizing_single_disease_search, fig.width=5, fig.height=3>>=
plot ( d1, name = "major depressive affective disorder, single episode, unspecified", type = "barplot" )
@

\noindent As a result, a barplot is obtained. The X axis contains the genes related to the disease of interest, sorted by the number of pubmed ids in which we can find the gene-disease association. alternatively, the results can be visualized for the diseases.\\

<<barplot_visualizing_single_gene_search, fig.width=5, fig.height=3>>=
plot ( t1, name = "NPY", type = "barplot" )
@

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section {Analyze the association between two diseases from the molecular perspective} 

\noindent We can study the association between two diseases from the point of view of shared genetic contribution. More precisely, we can estimate the degree of association of two diseases by means of the number of genes that are shared by the two diseases, over the total number of disease genes. Similarity measures such as the Jaccard Index can be used to estimate disease similarity. The significance of the Jaccard index is estimated by a bootstrap procedure (see below).\\

\subsection{Using the Jaccard Index}

\noindent The Jaccard Index, also known as the Jaccard similarity coefficient, is a statistic used for comparing the similarity of two  sets, and is defined as the size of the intersection divided by the size of the union of the sample sets:

\begin{equation*}
J(A, B) = \frac{\mid A \cap B \mid}{\mid A \cup B \mid}
\end{equation*}

\noindent \texttt{psygenet2r} comes with functions to compute the Jaccard Index as an estimation of the similarity of two diseases based on shared genes, given information retrieved from PsyGeNET. The user can compute the Jaccard Index using the function \texttt{jaccardEstimation}. This function accepts multiple inputs: 

\begin{enumerate}
  \item Using a list of genes of interest the function will compute the Jaccard Index between the set of genes and all the diseases in PsyGeNET.
<<ji_1>>=
genes_interest <- c("SLC6A4", "DRD2", "HTR1B", "PLP1", "TH", "DRD3")
ji1 <- jaccardEstimation(genes_interest, database = "ALL")
@
  \item Using a list of genes of interest and a list of diseases of interest, the function computes the Jaccard Index between the set of genes and each disease:
<<ji_2>>=
disease_interest <- 
  c("delirium", "bipolar i disorder", "severe depression", "cocaine addiction")
ji2 <- jaccardEstimation(genes_interest, disease_interest, database = "ALL")
@
  \item With a list of diseases of interest, the function will calculate the Jaccard Index between themselves:
<<ji_3>>=
ji3 <- jaccardEstimation(disease_interest, database = "ALL")
@
\end{enumerate}
%' 
%' \subsubsection*{According to multiple CUIs}
%' 
%' \noindent The following run will calculate the Jaccard Index between two diseases:
%' 
%' <<jaccard_cuid_1, cache=TRUE>>=
%' j1 <- jaccardEstimation( c("umls:C0005586", "umls:C0001973" ) )
%' j1
%' @
%' 
%' \subsubsection*{According to multiple disease}
%' 
%' \noindent The following execution will calculate the Jaccard Index between three diseases:
%' 
%' <<jaccard_muliple_diseases, cache=TRUE>>=
%' sel <- c( "Alcohol-Related Disorders", "Cocaine Abuse", 
%'           "Bipolar Depression" )
%' j2 <- jaccardEstimation( sel, database="ALL" )
%' j2
%' @
%' 
%' \subsubsection*{According to a gene list and a CUI, disease or psychiatric disorder}
%' 
%' \noindent The following run will calculate the Jaccard Index between a set of genes and a single diseases:
%' 
%' <<jaccard_genes, cache=TRUE>>=
%' j3 <- jaccardEstimation( c( "IL1B", "IL6", "TNF", "CXCL8", "CYP1A1",
%'                         "CYP1B1", "CSF2" ), "umls:C0005586" )
%' j3
%' @
%' 
%' \noindent The following code will calculate the Jaccard Index between a set of genes and two diseases:
%' 
%' <<jaccard_genes, cache=TRUE>>=
%' j4 <- jaccardEstimation( c( "IL1B", "IL6", "TNF", "CXCL8", "CYP1A1",
%'                         "CYP1B1", "CSF2" ), c("umls:C0005586", "umls:C0001973" ) )
%' j4
%' @

\noindent To determine if the association between two diseases as estimated by the Jaccard Index was statistically significant, we applied a bootstrap procedure to estimate the likelihood of obtaining a Jaccard Index greater than the one obtained for the association between the diseases by chance. In other words, we sampled at random gene sets of size n and p (n, p is the number of genes associated to disease 1 and 2, respectively) from a population of human disease genes obtained from DisGeNET \cite{REF3}. These random gene sets (n and p) were then used to compute the Jaccard Index for diseases 1 and 2. This procedure was repeated 100 times. Then, we calculated the number of times we obtained a Jaccard Index for the random gene sets larger than the observed value of the Jaccard Index. % We only kept disease associations for which we obtained less than 5/100 random cases an Jaccard Index value larger than the observed one.

\noindent The raw results are stored in \texttt{JaccardIndexPsy} and can be obtained using the function \texttt{extract}. For example:

<<ji1_extract>>=
head(extract(ji1))
tail(extract(ji1))
@

\subsection{Plotting results: Jaccard Index}

\noindent The plot of the result of a \texttt{jaccardEstimation} using a singe set of genes corresponds to a bar-plot of the Jaccard Index with each disease:

<<ji1_plot, fig.width=5, fig.height=3>>=
plot(ji1, cutOff = 0.1)
@

\noindent The previous bar-plot shows the Jaccard Index greater than 0.1 obtained from testing each diseases in PsyGeNET. When given a set of genes and a set of diseases, the resulting plot is equivalent:

<<ji2_plot, fig.width=5, fig.height=3>>=
plot(ji2)
@

\noindent The plot resulting from more than one disease is a heat-map with the given disease as X axis and all the diseases that share genes with them placed as Y axis. The intensity of the color represents the value of the Jaccard Index between, being the darker one the major Jaccard Index.\\

<<ji3_plot, fig.width=5, fig.height=3>>=
plot(ji3)
@

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Warnings}

\noindent \texttt{venn} and \texttt{heatmap} type arguments do not allow queries for single gene:

\begin{verbatim}
> plot( t1, type = "venn" )
==> Error: For this type of chart, a multiple gene query created with
 'psygenetGeneList' is required.

> plot( t1, type = "vennA" )
==> Error: For this type of chart, a multiple gene query created with
 'psygenetGeneList' is required.

> plot( t1, type = "heatmap" )
==> Error: For this type of chart, a multiple gene query created with
 'psygenetGeneList' is required.
\end{verbatim}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{thebibliography}{9}

\bibitem{REF1}
  Alba Gutierrez-Sacristan; Solene Grosdidier; Olga Valverde; Marta Torrens; Alex Bravo; Janet Pinero; Ferran Sanz; Laura I. Furlong. 
  \textbf{PsyGeNET: a knowledge platform on psychiatric disorders and their genes}
  Bioinformatics 2015
  doi: 10.1093/bioinformatics/btv301

\bibitem{REF2}
  Sullivan, Patrick F;  Daly, Mark J;  O'Donovan, Michael. 
  \textbf{Genetic architectures of psychiatric disorders: the emerging picture and its implications}
  Nature reviews. Genetics (2012) vol. 13 (8) p. 537-51

\bibitem{REF3}
  Janet Piñero, Núria Queralt-Rosinach, Àlex Bravo, Jordi Deu-Pons, Anna Bauer-Mehren, Martin Baron, Ferran Sanz, Laura I Furlong. 
  \textbf{DisGeNET: a discovery platform for the dynamical exploration of human diseases and their genes}
  Database (2015) Vol. 2015: article ID bav028; 
  doi:10.1093/database/bav028 

\bibitem{REF4}
  Bravo, À.; Piñero, J.; Queralt, N.; Rautschka, M.; Furlong, L.I. 
  \textbf{Extraction of relations between genes and diseases from text and large-scale data analysis: implications for translational research}
  BMC Bioinformatics 2015, 16:55
  doi:10.1186/s12859-015-0472-9

\bibitem{REF5}
  Flint, Jonathan;  Kendler, Kenneth S. 
  \textbf{The genetics of major depression}
  Neuron (2014) Vol. 81 (3) p. 484-503

\bibitem{Treutlein} 
  Jens Treutlein, Sven Cichon et al.
  \textbf{Genome-wide association study of alcohol dependence}. 
  Archives of general psychiatry
  (2009) vol.66(7) p.773
  doi: 10.1001/archgenpsychiatry.2009.83.
  
\end{thebibliography}   


\end{document}
